### PR TITLE
Adds Isthmus README.md

### PR DESCRIPTION
**Pull request by RCHowell**
Adds a basic README to Isthmus root — https://github.com/substrait-io/substrait-java/issues/44

### CODE REVIEW

**Review by JamesRTaylor — APPROVED** [review]
LGTM. Thanks for the contribution!

**Comment by RCHowell** [comment]
Thank you, James. As a first-time contributor, I believe this needs a manual workflow approval and squash+merge from a maintainer.

**Review by jacques-n — APPROVED** [review]
Thanks for the contribution @RCHowell !

**Pull request merged**
